.github/workflows/python-versions.yml

```diff
@@ -62,7 +62,8 @@ jobs:
   py_snowflake:
     runs-on: ubuntu-latest
 
-    if: github.repository == 'streamlit/streamlit'
+    # Won't run if PR from fork, since that PR wouldn't have secrets access
+    if: ${{ github.repository == 'streamlit/streamlit' && ( github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository ) }}
```

Comment:
```suggestion
    # Won't run if PR from fork or by dependabot, since that PR wouldn't have secrets access
    # See: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions
    if: | 
        github.repository == 'streamlit/streamlit' && (
        
        (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]') ||
        (github.event_name == 'push') ||
        (github.event_name == 'workflow_dispatch')
        )
```
Depedanbot also doesn't have access to secrets.

I also simplified the expression to reduce the number of negations. Now all cases where a step is run is explicitly specified.